parens_2.py

```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env python
+
+import sys
+
+
+def _parens(prefix, left, right):
+    if left == 0 and right == 0: sys.stdout.write(prefix)
+    if left > 0: _parens(''.join([prefix, '(']), left-1, right)
```

Comment:
Is this a common idiom? It looks much less readable to me.

Overall, this reduces performance vs the version that it changes:

```
$ time ./parens.py 15 > /dev/null

real	0m23.878s
```